parsers/FR.py

```diff
@@ -156,6 +156,10 @@ def fetch_production(
         if not any([is_not_nan_and_truthy(v) for k, v in production.items()]):
             continue
 
+        # only keep datapoints which minutes are 0 and 30 (ignore 15 and 45)
+        if arrow.get(row[1]["date_heure"]).datetime.minute not in [0,30]:
+            continue
+
```

Comment:
This would cause our data to become inaccurate.

A better solution would be to collect all data point and average 0 and 15 and then average 30 and 45 into datapoints for 0 and 30 respectively. That would allow us to get accurate but averaged data.

I'm not sure why this is needed though? The backend should average all data points in an hour to a hourly value?

Reply:
Right now when we refetch data, only half of the data is overwritten (the ones with minute 0 or minute 30) and the other half kept untouched. So we average the new data with old data

Reply:
this is needed because in the historical dataset we get a 1/2 hour granularity vs 1/4 hour for real-time data. This means that when we refetch we only correct 1/2 points and our dataset is not the same as the one published by RTE and we lose accuracy

Reply:
Ah okay that makes sense, but in order to keep all the new data we need to average it into two data points from four before we return it otherwise if we just skip it we would lose data for 2 datapoints.

For example if there is a sudden spike in say gas production during a 15 minute period that start at 12:15 we would lose that additional gas consumption and would be getting inaccurate values.

If we instead do
```python
datapoint_1 = (datapoint_1 + datapoint_2) / 2
```
we gett all of the data but still only two datapoint for each hour.

Reply:
See the new changes